File: bionic-js-swift/BjsTests/Bjs/UserOfToyComponent1.swift
import JavaScriptCore
import Bjs

class UserOfToyComponent1: BjsObject {
    
    static var lastToy: ToyComponent1? {
        get {
            return UserOfToyComponent1.bjs.getWrapped(bjs.getProperty(UserOfToyComponent1.bjsClass, "lastToy"), ToyComponent1.self)
        }
        set {
            UserOfToyComponent1.bjs.setProperty(UserOfToyComponent1.bjsClass, "lastToy", UserOfToyComponent1.bjs.putWrapped(newValue, ToyComponent1BjsWrapper.self))
        }
    }
    
    class func add(_ offset: Int?, _ int1: Int?, _ int2: Int?) -> Int? {
        return UserOfToyComponent1.bjs.getInt(bjs.call(UserOfToyComponent1.bjsClass, "add", UserOfToyComponent1.bjs.putPrimitive(offset),
                                           UserOfToyComponent1.bjs.putPrimitive(int1), UserOfToyComponent1.bjs.putPrimitive(int2)))
    }
    
    class func getToy(_ int1: Int?, _ int2: Int?) -> ToyComponent1? {
        return UserOfToyComponent1.bjs.getWrapped(bjs.call(UserOfToyComponent1.bjsClass, "getToy", UserOfToyComponent1.bjs.putPrimitive(int1),
                                              UserOfToyComponent1.bjs.putPrimitive(int2)), ToyComponent1.self)
    }
    
    class func getSum(_ toy1: ToyComponent1?, _ toy2: ToyComponent1?) -> Int? {
        return UserOfToyComponent1.bjs.getInt(bjs.call(UserOfToyComponent1.bjsClass, "getSum",
                                           UserOfToyComponent1.bjs.putWrapped(toy1, ToyComponent1BjsWrapper.self),
                                           UserOfToyComponent1.bjs.putWrapped(toy2, ToyComponent1BjsWrapper.self)))
    }
        
    class func bjsFactory(_ jsObject: JSValue) -> UserOfToyComponent1 {
        return UserOfToyComponent1(jsObject)
    }
    
    private static var _bjsLocator = BjsLocator("TestProject", "UserOfToyComponent1")
    override class var bjsLocator: BjsLocator { _bjsLocator }
}

File: bionic-js-runtime-java/src/test/java/acceptance/toy1/bjs/UserOfToyComponent1.java
package acceptance.toy1.bjs;

import bionic.js.Bjs;
import bionic.js.BjsObject;
import bionic.js.BjsObjectTypeInfo;
import bionic.js.BjsTypeInfo;
import jjbridge.api.runtime.JSReference;

@BjsTypeInfo.BjsLocation(project = "TestProject", module = "UserOfToyComponent1")
public class UserOfToyComponent1 extends BjsObject
{
    public static final Bjs.JSReferenceConverter<UserOfToyComponent1> bjsFactory = UserOfToyComponent1::new;
    public static final Bjs bjs = BjsObjectTypeInfo.get(UserOfToyComponent1.class).bjsLocator.get();
    private static final JSReference bjsClass = BjsObjectTypeInfo.get(UserOfToyComponent1.class).bjsClass();

    protected <T extends BjsObject> UserOfToyComponent1(Class<T> type, JSReference jsObject)
    {
        super(type, jsObject);
    }

    protected <T extends BjsObject> UserOfToyComponent1(Class<T> type, JSReference[] arguments)
    {
        super(type, arguments);
    }

    public UserOfToyComponent1(JSReference jsObject)
    {
        this(UserOfToyComponent1.class, jsObject);
    }

    public static ToyComponent1BjsExport lastToy()
    {
        return bjs.getWrapped(bjs.getProperty(bjsClass, "lastToy"));
    }

    public static void lastToy(ToyComponent1BjsExport value)
    {
        bjs.setProperty(bjsClass, "lastToy", bjs.putWrapped(value, ToyComponent1BjsExport.Wrapper.class));
    }

    public static Integer add(Integer offset, Integer int1, Integer int2)
    {
        return bjs.getInteger(bjs.call(bjsClass, "add", bjs.putPrimitive(offset), bjs.putPrimitive(int1),
                bjs.putPrimitive(int2)));
    }

    public static ToyComponent1BjsExport getToy(Integer int1, Integer int2)
    {
        return bjs.getWrapped(bjs.call(bjsClass, "getToy", bjs.putPrimitive(int1), bjs.putPrimitive(int2)));
    }

    public static Integer getSum(ToyComponent1BjsExport toy1, ToyComponent1BjsExport toy2)
    {
        return bjs.getInteger(bjs.call(bjsClass, "getSum", bjs.putWrapped(toy1, ToyComponent1BjsExport.Wrapper.class),
                bjs.putWrapped(toy2, ToyComponent1BjsExport.Wrapper.class)));
    }

    public static Integer getSum2(ToyComponent2BjsExport toy1, ToyComponent1BjsExport toy2)
    {
        return bjs.getInteger(bjs.call(bjsClass, "getSum2", bjs.putWrapped(toy1, ToyComponent2BjsExport.Wrapper.class),
                bjs.putWrapped(toy2, ToyComponent1BjsExport.Wrapper.class)));
    }

    public static Boolean isLastToy(ToyComponent1BjsExport toy)
    {
        return bjs.getBoolean(bjs.call(bjsClass, "isLastToy", bjs.putWrapped(toy, ToyComponent1BjsExport.Wrapper.class)));
    }

    public static Double piSum()
    {
        return bjs.getDouble(bjs.call(bjsClass, "piSum"));
    }

    public static Integer additionalValue(ToyComponent2BjsExport toy)
    {
        return bjs.getInteger(bjs.call(bjsClass, "additionalValue", bjs.putWrapped(toy, ToyComponent2BjsExport.Wrapper.class)));
    }
}
